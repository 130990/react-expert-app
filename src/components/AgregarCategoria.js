import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AgregarCategoria = ({ setCategorias }) => {
    const [inputValue, setInputValue] = useState('');
    const handleCambiarInput = (e) => {        
        setInputValue(e.target.value)
    }
    const handlEnviar = (e) => {
        e.preventDefault();
        console.log('Llamado Submit......');
        if (inputValue.length > 2) {
            setCategorias(cate => [inputValue,...cate])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handlEnviar}>
            <input type="text" value={inputValue} onChange={handleCambiarInput} />
        </form>
    )
}

AgregarCategoria.propTypes ={
    setCategorias: PropTypes.func.isRequired
}
