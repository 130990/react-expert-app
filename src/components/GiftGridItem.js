import React from 'react'
import '../index.css';
import PropTypes from 'prop-types'

export const GiftGridItem = ({title, url }) => {
    return (
        <div className="card animate__animated animate__fadein">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes ={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
