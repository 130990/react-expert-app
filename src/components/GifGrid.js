import React from 'react'
import { useFecthGifs } from '../hooks/useFecthGifs'
import { GiftGridItem } from './GiftGridItem'

export const GifGrid = ({ categoria }) => {

    const { data: imagenes, loading } = useFecthGifs(categoria);


    return (
        <>
            <h3>{categoria}</h3>
            {loading &&<p className='animate__animated animate__flash'>Cargando imagenes...</p> }
            <div className="card-grid" >
                {
                    imagenes.map((img) => (
                        <GiftGridItem key={img.id} {...img} /> //enviando las imagenes con el spread, permite enviar las propiedades como un solo objeto, q puede desestructurase del otro lado
                    ))
                }
            </div>
        </>
    )
}
