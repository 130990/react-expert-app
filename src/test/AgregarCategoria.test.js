import React from 'react';
import { shallow } from 'enzyme';
import { AgregarCategoria } from '../components/AgregarCategoria'
import '@testing-library/jest-dom'


describe('Pruebas del componente AgregarCategoria', () => {
    const setCategorias = jest.fn();
    let ResultadoObtenido = shallow(< AgregarCategoria setCategorias={setCategorias} />)

    beforeEach(() => {
        ResultadoObtenido = shallow(< AgregarCategoria setCategorias={setCategorias} />)
    })

    test('Debe de mostrarse correctamente ', () => {
        expect(ResultadoObtenido).toMatchSnapshot();
    })
    test('Debe de cambiar la caja de texto ', () => {
        const input = ResultadoObtenido.find('input');
        const valor = 'Prueva texto';
        input.simulate('change', { target: { value: valor } })
    })

    test('NO Debe de simular el "submit" del formulario', () => {
        const input = ResultadoObtenido.find('form');
        input.simulate('submit', { preventDefault() { } }) // esto preventDefault(){} es igual a preventDefault: () =>{}

        expect(setCategorias).not.toHaveBeenCalled();
    })

    test('Debe de llamar al setCategorias y limpiar la caja de texto', () => {
        const valor = 'Prueva texto';
        ResultadoObtenido.find('input').simulate('change', { target: { value: valor } });
        ResultadoObtenido.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategorias).toHaveBeenCalled();
        expect(ResultadoObtenido.find('input').prop('value')).toBe('');




    })



})
