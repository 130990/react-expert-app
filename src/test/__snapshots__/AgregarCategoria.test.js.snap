// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente AgregarCategoria Debe de mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
